src/Symfony/Component/DependencyInjection/Compiler/ResolveBindingsPass.php

```diff
@@ -214,4 +231,11 @@ private function getBindingValue(BoundArgument $binding)
 
         return $bindingValue;
     }
+
+    private function searchBindingByPropertyName(array $bindings, string $propertyName)
+    {
+        return array_filter($bindings, function ($key) use ($propertyName) {
+            return false !== strpos($key, $propertyName);
```

Comment:
shouldn't we have an exact match instead of strpos? this might trigger a false positive.